Swift/Database.swift

```diff
@@ -118,6 +118,35 @@ public final class Database {
     }
     
     
+    /// Saves a document to the database. When write operations are executed concurrently,
+    /// and if conflicts occur, conflict handler will be called. Use the handler to directly
+    /// edit the document. Returning true, will save the document. Returning false, will cancel
+    /// the save operation.
+    ///
+    /// - Parameters:
+    ///   - document: The document.
+    ///   - conflictHandler: The conflict handler closure which can be used to resolve it.
+    /// - Returns: True if successful. False on failure
+    /// - Throws: An error on a failure.
+    @discardableResult public func saveDocument(
+        _ document: MutableDocument, conflictHandler: @escaping (MutableDocument, Document?) -> Bool
+        ) throws -> Bool {
+        do {
+            let conflictHandler = { (current: CBLMutableDocument, old: CBLDocument?) -> Bool in
+                if let oldDoc = old {
+                    return conflictHandler(MutableDocument(current), Document(oldDoc))
+                }
+                return conflictHandler(MutableDocument(current), nil)
+            }
+            
+            try _impl.save(document._impl as! CBLMutableDocument, conflictHandler: conflictHandler)
+            return true
+        } catch let err as NSError {
+            throw err
```

Comment:
The logic here should be the same as the saveDocument() above.

Reply:
doubt: if it is already handling the conflict, is there a chance for getting the `CBLErrorConflict`? exceptions inside conflict handler will also not return the CBLErrorConflict right.